internal/db/sqlite/sql/migrations/folder/03-drop-bad-invalid.sql

```diff
@@ -0,0 +1,17 @@
+-- Copyright (C) 2025 The Syncthing Authors.
+--
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this file,
+-- You can obtain one at https://mozilla.org/MPL/2.0/.
+
+-- Remove broken file entries in the database.
+DELETE FROM files
+    WHERE type == 0 -- files
+        AND NOT deleted -- that are not deleted
+        AND blocklist_hash IS null -- with no blocks
+        AND local_flags & {{.LocalInvalidFlags}} == 0 -- and not invalid
+;
+
+-- Force a new index transmission.
+DELETE FROM indexids
```

Comment:
Afaik this only affected the local device. And after removing those local files above, they'll be scanned and sent with the existing index, while the old, "incorrect" entries are gone. As in there should be no need to drop/retransmit all index info.

Reply:
Ah no, the code that clobbered the invalid bit is on receiving indexes... Then of course retransmission is indeed needed.

I really don't get the full picture here.